### PR TITLE
Further Category Slidedown Src Changes (Merge to Feature)

### PR DESCRIPTION
**Pull request by rgomezp**
# Description
The branch `cat-slidedown-2ndpass` includes all the changes requested in PR #658 to facilitate viewing the differences in this specific pass.

# Changes
- Minor refactoring and logic tweaks
- Renamed tag methods in `TagManager` from
   - `syncTags` to `sendTags`
   - `downloadTags` to `getTags`
- Deep copy of tags object in utils methods
   - `convertTagBooleanValuesToNumbers`
   - `convertTagNumberValuesToBooleans`
- SCSS changes:
   - added specific `onesignal-category-label-input` class for `input` in `label`
   - refactored to make SCSS more concise, avoid code duplicatiion
   - Added back legacy class names with "popover" instead of "slidedown" for backwards-compatibility
- `getDomElementOrStub` logging change to `info` type
- Prevents actual API call to update tags iff there was no change to category configuration detected
- Failed sendTags call (e.g: bad connection) results in error message shown

# Tested
- HTTPS
   - [x] Chrome
   - [x] Firefox
   - [x] Safari 

- HTTPS with Workaround 
   - [x] Firefox
   - [x] Chrome
   - [x] Safari 
      
# Screenshot
### Support for old legacy "popover" selectors....
Screen Shot 2020-06-29 at 3 45 37 PM

### Category Slidedown
Screen Shot 2020-06-29 at 3 59 03 PM

# To do items
- [x] Error message styling
- [ ] Fix bundle size issues

---
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-website-sdk/660)

### CODE REVIEW

File: src/helpers/EventHelper.ts [hunk 1]
```diff
@@ -40,10 +40,23 @@ export default class EventHelper {
 
   static async _onSubscriptionChanged(newSubscriptionState: boolean | undefined) {
     EventHelper.onSubscriptionChanged_showWelcomeNotification(newSubscriptionState);
+    EventHelper.onSubscriptionChanged_sendCategorySlidedownTags(newSubscriptionState);
     EventHelper.onSubscriptionChanged_evaluateNotifyButtonDisplayPredicate();
     EventHelper.onSubscriptionChanged_updateCustomLink();
   }
 
+  private static async onSubscriptionChanged_sendCategorySlidedownTags(isSubscribed: boolean | undefined) {
+    if (isSubscribed === true) {
```
Comment (jkasten2) [inline src/helpers/EventHelper.ts:49]: Could be an early return .

File: src/helpers/EventHelper.ts [hunk 2]
```diff
@@ -40,10 +40,23 @@ export default class EventHelper {
...
+        const promptOptions = await OneSignal.context.appConfig.userConfig.promptOptions;
+        const isUsingCategoryConfig = !!promptOptions && !!promptOptions.slidedown &&
```
Comment (jkasten2) [inline src/helpers/EventHelper.ts:51]: Should check for tag length like below.

Also should be a shared method so this doesn't have to be duplicated code.

File: src/utils/TagUtils.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import Log from '../libraries/Log';
 import { TagsObject } from '../models/Tags';
+import _ from "lodash";
```
Comment (jkasten2) [inline src/utils/TagUtils.ts:3]: Need remove and use something very small or write our own code for what we use here.

File: src/helpers/EventHelper.ts (hunk 2 above)
Comment (rgomezp) [reply src/helpers/EventHelper.ts]: Done.

File: src/helpers/EventHelper.ts (hunk 1 above)
Comment (rgomezp) [reply src/helpers/EventHelper.ts]: Done.

**Pull request merged**